### DISCUSSION TITLE
How to get the size of a ft.Control?

### DISCUSSION THREAD

**Discussion by wschume** [flet-dev/flet#3040] [Q&A] [answered]
### Question

Is there a way in flet to get the size of an `ft.Control`? A page has the [resize](https://flet.dev/docs/controls/page#on_resize) event, which you can access and get the changed size of the page. But for controls, the `width` and `height` properties remain `None` unless you set them explicitly.

The use case is to have multiple `ft.Row`s in a `ft.Column`, and I would like to limit the space acquired by the rows children to be limited. In essence I am trying to build a table but cannot use `ft.DataTable` for this, as the rows have various heights, I don't really have a header, etc. Also `ft.GridView` is not the right choice IMHO.

Thank you in advance for your help.

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment, marked as answer]
https://python.plainenglish.io/how-to-determine-the-size-of-any-flet-control-python-ddc5631dd442

**Reply by wschume** [reply]
Very nice. I now recall, that I read this article before and just forgotten about it.
@ndonkoHenri May I recommend to modularize this approach?

If it is not desired, that any control has the `on_resize` event, then a dedicated control could be implemented that has nothing of the `Canvas` stuff in it (like the [`shapes`](https://flet.dev/docs/controls/canvas#shapes) property etc.

I somehow feel like the suggested solution (though it currently solves my problem) somehow feels a little hacky...
None the less, I will mark you answer as the current solution to my question.
